Src/World/Timing.cpp

```diff
@@ -1,70 +1,78 @@
 #include "Timing.h"
 
-#include <Math/Math.h>
-
 
 // The maximum amount of time the accumulator can store.
-constexpr double MAX_ACCUMULATED_SECONDS = 3.0;
+static constexpr double MAX_ACCUMULATED_SECONDS = 3.0;
+using Clock = std::chrono::high_resolution_clock;
+
 
-Timing::Timing(int tickrate) {
-    timeStep = 1.0 / tickrate;
-    accumulatedSeconds = 0.0;
-    secondToLastFixedTime = 1.0;
-    lastFixedTime = 1.0;
+Timing::Timing(int tickrate, int framerate) {
+    setTickRate(tickrate-1);
+    setFrameRate(framerate-1);
+    sinceLastTick = 0.f;
+    sinceLastFrame = 0.f;
+    avgTickRate = 0.f;
+    avgFrameRate = 0.f;
 
     initialTime = std::chrono::high_resolution_clock::now();
-    prevTime = initialTime;
+    compareTick = initialTime - tickStepDuration; // runFirst
+    compareFrame = initialTime - frameStepDuration; // runFirst
+    lastTick = compareTick;
+    lastFrame = compareFrame;
 }
 
-double Timing::getTimeStep() const {
-    return timeStep;
+void Timing::setFrameRate(int rate) {
+    frameRate = rate;
+    frameStep = ceil(1000000000.f / float(rate));
+    frameStepDuration = std::chrono::nanoseconds(frameStep);
 }
-
-void Timing::addSecondsToAccumulator(double seconds) {
-    if (seconds <= 0.0) { return; }
-    accumulatedSeconds += seconds;
-
-    if (accumulatedSeconds > MAX_ACCUMULATED_SECONDS) {
-        accumulatedSeconds = MAX_ACCUMULATED_SECONDS;
-    }
+void Timing::setTickRate(int rate) {
+    tickRate = rate;
+    tickStep = ceil(1000000000.f / float(rate));
```

Comment:
This runs into extreme floating-point inaccuracies. The exponent required to store anything close to this number is huge, the result is effectively an integer, which is only usable in this case by chance. This needs to be changed.

Reply:
tickStep is the (integer) number of nanoseconds between frames, with 1,000,000,000 (1billion) being 1 second.
Fractions can be effectively thrown out.
I'll look up some integer size charts and do this properly, dependent upon intentions and goals for timing systems.